feeds/wlan-ap/opensync/src/platform/openwrt/src/rrm/src/rrm_channel.c

```diff
@@ -122,58 +169,52 @@ void rrm_nf_timer_handler(struct ev_loop *loop, ev_timer *timer, int revents)
 	{
 		noise = 0;
 		rrm_config = NULL;
-		nf_drop_threshold = 0;
 
 		if (ubus_get_noise(radio->config.if_name, &noise))
 			continue;
 
 		nf = (int32_t)noise;
 
-		if (nf > -1 || nf < -120)
+		if ((nf > -10) || (nf < -120))
 			continue;
 
 		rrm_config = rrm_get_rrm_config(radio->config.type);
 
 		if (rrm_config == NULL)
 			continue;
 
-		if (nf < rrm_config->rrm_data.noise_lwm )
-		{
-			rrm_config->rrm_data.noise_lwm = nf;
-			LOGD("[%s] noise_lwm set to %d", radio->config.if_name, nf);
+		if (rrm_config->rrm_data.backup_channel == 0)
```

Comment:
Do we also need a test to make sure we are not already on the backup channel?

I see at line 228 that we reset the data after a channel switch, but we could still accumulate new data and try to switch again.

Reply:
Good point. Thanks for bringing this up.  This check might be required.
I was thinking to not even collect the nf samples if the configured channel is same as backup channel.
So, we can add a check at the top of this function like below,

		if ((rrm_config->rrm_data.backup_channel == 0) || 
                     (rrm_config->rrm_data.backup_channel == ConfiguredChannel))
			continue;

Reply:
done